x/exchange/client/cli/helpers.go

```diff
@@ -0,0 +1,282 @@
+// then uses the provided maker to make the Msg that it then generates or broadcasts as a Tx.
+//
+// R is the type of the Msg.
+func genericTxRunE[R sdk.Msg](maker msgMaker[R]) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		clientCtx, err := client.GetClientTxContext(cmd)
+		if err != nil {
+			return err
+		}
+
+		flagSet := cmd.Flags()
+		msg, err := maker(clientCtx, flagSet, args)
+		if err != nil {
+			return err
+		}
+
+		cmd.SilenceUsage = true
+		return tx.GenerateOrBroadcastTxCLI(clientCtx, flagSet, msg)
+	}
+}
+
+// govTxRunE returns a cobra.Command.RunE function that gets the client.Context and FlagSet,
+// then uses the provided maker to make the Msg. The Msg is then put into a governance
+// proposal and either generated or broadcast as a Tx.
+//
+// R is the type of the Msg.
+func govTxRunE[R sdk.Msg](maker msgMaker[R]) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		clientCtx, err := client.GetClientTxContext(cmd)
+		if err != nil {
+			return err
+		}
+
+		flagSet := cmd.Flags()
+		msg, err := maker(clientCtx, flagSet, args)
+		if err != nil {
+			return err
+		}
+
+		cmd.SilenceUsage = true
+		return govcli.GenerateOrBroadcastTxCLIAsGovProp(clientCtx, flagSet, msg)
+	}
+}
+
+// queryReqMaker is a function that creates a query request message.
+//
+// R is the type of request message.
+type queryReqMaker[R any] func(clientCtx client.Context, flagSet *pflag.FlagSet, args []string) (*R, error)
+
+// queryEndpoint is a grpc_query endpoint function.
+//
+// R is the type of request message.
+// S is the type of response message.
+type queryEndpoint[R any, S proto.Message] func(queryClient exchange.QueryClient, ctx context.Context, req *R, opts ...grpc.CallOption) (S, error)
+
+// genericQueryRunE returns a cobra.Command.RunE function that gets the query context and FlagSet,
+// then uses the provided maker to make the query request message. A query client is created and
+// that message is then given to the provided endpoint func to get the response which is then printed.
+//
+// R is the type of request message.
+// S is the type of response message.
+func genericQueryRunE[R any, S proto.Message](reqMaker queryReqMaker[R], endpoint queryEndpoint[R, S]) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		clientCtx, err := client.GetClientQueryContext(cmd)
+		if err != nil {
+			return err
+		}
+
+		req, err := reqMaker(clientCtx, cmd.Flags(), args)
+		if err != nil {
+			return err
+		}
+
+		cmd.SilenceUsage = true
+		queryClient := exchange.NewQueryClient(clientCtx)
+		res, err := endpoint(queryClient, cmd.Context(), req)
+		if err != nil {
+			return err
+		}
+
+		return clientCtx.PrintProto(res)
+	}
+}
+
+// MarkFlagsRequired marks the provided flags as required and panics if there's a problem.
```

Comment:
Should this be with the rest of the flag functions?

Reply:
I waffled on this pretty hard. I think I ultimately put it in `helpers.go` because it was so generic. The stuff in `flags.go` was more related to adding, reading, and parsing specific flags. But really, the line between `flags.go` and `helpers.go` is pretty vague anyway. E.g. I thought about putting the string parsing stuff in `helpers.go`, but ultimately decided to leave them in `flags.go` so that they're right next to the flag reading functions that the parsers were designed for.

There were several times where I thought about moving this into `flags.go` but just decided I didn't care either way, so left it where I originally put it.

But, now that you say something, I will move it over.